test/e2e/framework/service/jig.go

```diff
@@ -390,8 +393,56 @@ func (j *TestJig) waitForAvailableEndpoint(timeout time.Duration) error {
 
 	go controller.Run(stopCh)
 
+	// If EndpointSlice API is enabled, then validate if appropriate EndpointSlice objects were also create/updated/deleted.
+	if _, err := j.Client.Discovery().ServerResourcesForGroupVersion(discoveryv1beta1.SchemeGroupVersion.String()); err == nil {
+		var esController cache.Controller
+		_, esController = cache.NewInformer(
+			&cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					obj, err := j.Client.DiscoveryV1beta1().EndpointSlices(j.Namespace).List(context.TODO(), metav1.ListOptions{
```

Comment:
It seems I wasn't explicit enough - it's not what I meant.

I meant:
```
options.LabelsSelector = "kubernetes.io/service-name=" + j.Name
obj, err := c.Client.DiscoveryV1beta1().EndpointSlices(j.Names0ace).List(context.TODO(), options)
...


Same below

Reply:
ups

Reply:
I need more coffee :upside_down_face: